Adding a script to check for line endings (valid UNIX ending)

kushalsingh007 opened the pull request:
-Created a bash script using grep to check for carriage return at line endings.
-Fix for issue #2281.

kushalsingh007 commented:
Yet again the build fail is due to Travis being flaky , kindly re-run the build again :)

authmillenon commented:
Actually it fails because there are trailing whitespaces in your script. Please remove them ;)

authmillenon commented:
(It's flaky when some random board fails. The static tests are quite stable)

Speaking of: could you add your script to them (add a call to your script to `dist/tools/travis-scripts/build_and_test.sh`). Also, I think it would be best if it only applies to new and changed files. Check out the other check scripts as for example `dist/tools/whitespacecheck/check.sh` or `dist/tools/cppcheck/check.sh` for how to do this.

kushalsingh007 commented:
Ok , I will have a look at it and make sure that the check applies only to new files :)

kushalsingh007 commented:
@authmillenon : I have added a new script which checks only for the newly added files and added a call to it in build_and_run.sh. However  I do not have the permission to modify build_and_run.sh ( and that is why the static case fails ) . So have a look at it .

authmillenon commented on dist/tools/travis-scripts/build_and_test.sh:
```diff
@@ -45,6 +45,15 @@ then
+       ./dist/tools/endingcheck/check.sh master --diff-filter=AC
+       RESULT=$(set_result $? $RESULT)
+
+       ./dist/tools/endingcheck/check.sh riot/master
```
One call of your script with `riot/master` (the current master at RIOT-OS/RIOT) and `--diff-filter=MA` (only look at modified [M] and added [A] files) should suffice (source: http://git-scm.com/docs/git-diff).

authmillenon commented:
> However I do not have the permission to modify build_and_run.sh ( and that is why the static case fails ). 

What do you mean? You obviously modified it (see my comment above). So how do you come to the impression you don't have any permission to do so?  :confused:

kushalsingh007 commented:
That's because it is the reason build failed

authmillenon commented:
Ah... Your script is not executable. That's why Travis is saying the permissions were denied (because it has no permission to execute it). Git has an understanding of UNIX permission flags so you have to give your script execution permission:

```bash
chmod +x ./dist/tools/endingcheck/check.sh
```

authmillenon commented:
(and than of course add this change to git ;) 

```bash
git add ./dist/tools/endingcheck/check.sh
git commit
```

kushalsingh007 commented:
Done , the changes and the build

authmillenon commented on dist/tools/endingcheck/check.sh:
```diff
@@ -0,0 +1,52 @@
+#
+
+BRANCH=${1}
+FILEREGEX='\.([sScHh]|cpp)$'
```
I would actually include all files, when I think about it, not just source code.

kushalsingh007 replied:
There are things like images which should be excluded from it.

authmillenon commented:
Please squash only when the review is done, otherwise it can get a little bit confusing with bigger changes ;-)

kushalsingh007 commented:
Ok , I'll keep that in mind from next time. So , can I squash now  to change the file regex ?

authmillenon commented:
No, just mark the commits as squashable (I usually use a `[SQUASH ME]` tag in front of the commit description) and squash it, if you get the ACK ;-)

authmillenon commented:
> (I usually use a [SQUASH ME] tag in front of the commit description)

See #2546 for example

gebart commented on dist/tools/travis-scripts/build_and_test.sh:
```diff
@@ -45,6 +45,11 @@ then
         ./dist/tools/externc/check.sh master
         RESULT=$(set_result $? $RESULT)
 
+       RESULT=$(set_result $? $RESULT)
```
Why is this duplicate line here?

gebart commented on dist/tools/travis-scripts/build_and_test.sh:
```diff
@@ -45,6 +45,11 @@ then
+       RESULT=$(set_result $? $RESULT)
+
+       ./dist/tools/endingcheck/check.sh riot/master --diff-filter=MA
+       RESULT=$(set_result $? $RESULT)
```
Increase indentation by 1 space

kushalsingh007 commented:
@OlegHahm : Done squashing.
